# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/Device.h
# Commit: f214290956c1
# Full Hash: f214290956c1c17ab7da33be0868521198ad3112
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 2
# Description:
#   Bug 1838693 Part 2: Rationalize GPUDevice lost promise handling. r=webgpu-reviewers,nical
#   
#   This ensures that both internal and external triggers of "lose the
#   device" resolve the lost promise using the same code path.
#   
# ==============================================================================

diff -r 843c7d19787b -r f214290956c1 dom/webgpu/Device.h
--- a/dom/webgpu/Device.h	Mon Oct 16 15:31:59 2023 +0000
+++ b/dom/webgpu/Device.h	Mon Oct 16 15:32:00 2023 +0000
@@ -105,6 +105,7 @@
   void GenerateValidationError(const nsCString& aMessage);
 
   bool IsLost() const;
+  bool IsBridgeAlive() const;
 
  private:
   ~Device();
@@ -121,7 +122,6 @@
   void GetLabel(nsAString& aValue) const;
   void SetLabel(const nsAString& aLabel);
   dom::Promise* GetLost(ErrorResult& aRv);
-  dom::Promise* MaybeGetLost() const { return mLostPromise; }
   void ResolveLost(Maybe<dom::GPUDeviceLostReason> aReason,
                    const nsAString& aMessage);
 